Added try/except statement for units config

kb1lqc opened the pull request:
When creating appropriate items based on proxy.ini configuration file
sections a new try/except statement catches the
ConfigParser.NoSectionsError. When the exception is caught it logs the
error and exits python.

This addresses and fixes issue #132.

kb1lqc committed:
Added try/except statement for units config

When creating appropriate items based on proxy.ini configuration file
sections a new try/except statement catches the
ConfigParser.NoSectionsError. When the exception is caught it logs the
error and exits python.

el-iso reviewed (commented):
This code looks good to me. Provided the only issue was the "NoSectionError" described in the original issue ticket, this should make proxy fail more gracefully.

Will this make it obvious to the user what they did wrong? 

I'll let someone else give final approval since I am not very familiar with this particular issue.

reillyeon commented on faraday/proxy.py:
```diff
@@ -581,7 +581,12 @@ def main():
-    units = callsign2COM()
+    try:
+        units = callsign2COM()
+
```
Style nitpick: Don't put a space between the `try` and `catch` blocks since they are logically connected, at least when the `try` block is only one line.

kb1lqc replied:
Fixed in https://github.com/FaradayRF/Faraday-Software/pull/193/commits/5db3383d59ad59ebe48530e1cf210c0b5c2a5334

reillyeon commented on faraday/proxy.py:
```diff
@@ -581,7 +581,12 @@ def main():
+
+    except ConfigParser.NoSectionError as e:
+        logging.error(e)
+        sys.exit()
```
On error Python should exit with a non-zero error code so when catching errors like this we should use `sys.exit(1)`. (Coming up with a scheme where 1 means configuration error, 2 means something else, etc. is entirely optional and more work than really necessary.)

kb1lqc replied:
Updated in https://github.com/FaradayRF/Faraday-Software/pull/193/commits/49a24e73af1a7b220b169340dc2ada718ce64fae

Picking status code 1 = configuration error. Where do we think this should be documented? I'm thinking either in `contributions.md` or a separate `errors.md` type file?

reillyeon reviewed (changes requested):
If main() throws an exception the application will print it out and exit anyways. What is the purpose of this?

reillyeon commented:
Ignore that comment about why the try/catch is needed at all. I forgot to delete it before submitting the review.

kb1lqc committed:
Removed Space Between try/except per @reillyeon

Decent style time. I'll take it.

kb1lqc committed:
Exit With Status Code 1

Per @reillyeon exiting with a non-zero value. Arbitrarily picking status
code 1 to be any configuration error.

kb1lqc committed:
Fixed PEP8 Inline Comment Error

Thought I had correctly implemented the inline comment on the fly. I was
wrong. Quick and easy fix.

kb1lqc commented:
@reillyeon pushed commit https://github.com/FaradayRF/Faraday-Software/pull/193/commits/3cb1b97ea4387e1bb8190f40f94335ffc95dc616 after you approved which fixed a PEP8 failure I let through. Letting Travic-CI catch up before merging. Thanks!
